File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/entitiesvb/vb/createcustomactivityentity2.vb
' <snippetcreatecustomactivityentity2>


 Dim prefix As String = "new_"

 Dim customEntityName As String = prefix &amp; "instantmessage"

 ' Create the custom activity entity.
 Dim request As CreateEntityRequest = New CreateEntityRequest With {
  .HasNotes = True,
  .HasActivities = False,
  .PrimaryAttribute = New StringAttributeMetadata With {
   .SchemaName = "Subject",
   .RequiredLevel = New AttributeRequiredLevelManagedProperty(AttributeRequiredLevel.None),
   .MaxLength = 100,
   .DisplayName = New Label("Subject", 1033)},
  .Entity = New EntityMetadata With {
   .IsActivity = True,
   .SchemaName = customEntityName,
   .DisplayName = New Label("Instant Message", 1033),
   .DisplayCollectionName = New Label("Instant Messages", 1033),
   .OwnershipType = OwnershipTypes.UserOwned,
   .IsAvailableOffline = True}}

 _serviceProxy.Execute(request)

 'Entity must be published

' </snippetcreatecustomactivityentity2>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/entities/cs/createcustomactivityentity2.cs
// <snippetcreatecustomactivityentity2>


 String prefix = "new_";

 String customEntityName = prefix + "instantmessage";

 // Create the custom activity entity.
 CreateEntityRequest request = new CreateEntityRequest
 {
     HasNotes = true,
     HasActivities = false,
     PrimaryAttribute = new StringAttributeMetadata
     {
         SchemaName = "Subject",
         RequiredLevel = new AttributeRequiredLevelManagedProperty(AttributeRequiredLevel.None),
         MaxLength = 100,
         DisplayName = new Label("Subject", 1033)
     },
     Entity = new EntityMetadata
     {
         IsActivity = true,
         SchemaName = customEntityName,
         DisplayName = new Label("Instant Message", 1033),
         DisplayCollectionName = new Label("Instant Messages", 1033),
         OwnershipType = OwnershipTypes.UserOwned,
         IsAvailableOffline = true,

     }
 };

 _serviceProxy.Execute(request);

 //Entity must be published

// </snippetcreatecustomactivityentity2>